compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUtils.cpp

```diff
@@ -298,34 +287,44 @@ bool isMatmul(linalg::LinalgOp linalgOp) {
          match6DLinalgGenericMatmul(linalgOp);
 }
 
+/// Utility to identify if the input operand has matmul-like op in its
+/// def-chain.
+bool isMatmulInDefChain(Value operand) {
+  if (isa<BlockArgument>(operand)) return false;
+  while (Operation *defOp = operand.getDefiningOp()) {
+    if (isa<arith::ConstantOp>(defOp)) return false;
+    if (auto defLinalgOp = dyn_cast_or_null<linalg::LinalgOp>(defOp)) {
+      if (isMatmul(defLinalgOp)) {
+        return true;
+      }
+      break;
+    }
+    if (auto forOp = dyn_cast_or_null<scf::ForOp>(defOp)) {
+      bool findMatmul = false;
+      forOp.getBody()->walk<WalkOrder::PostOrder, ReverseIterator>(
+          [&](linalg::LinalgOp op) {
+            if (isMatmul(op)) {
+              findMatmul = true;
+              return WalkResult::interrupt();
+            }
+            return WalkResult::advance();
+          });
+      if (findMatmul) return true;
+    }
+    operand = defOp->getOperand(0);
+    if (isa<BlockArgument>(operand)) return false;
```

Comment:
No need for this. The control is flowing to the beginning of the `while` and the check there will anyway take care of it.